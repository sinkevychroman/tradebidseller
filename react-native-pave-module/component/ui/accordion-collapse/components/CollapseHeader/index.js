import React from 'react';
import {View} from 'react-native';

const CollapseHeader = React.forwardRef(({children, ...restProps}, ref) => {
  return (
    <View ref={ref} {...restProps}>
      {children}
    </View>
  );
});

export default CollapseHeader;
