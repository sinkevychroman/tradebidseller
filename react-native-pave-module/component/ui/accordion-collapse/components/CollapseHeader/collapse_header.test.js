import 'react-native';
import React from 'react';

import CollapseHeader from './index';
import renderer from 'react-test-renderer';
import {Text, View} from 'react-native';

describe('Collapse Header', () => {
  it('Should render CollapseHeader', () => {
    renderer.create(<CollapseHeader>Test</CollapseHeader>);
  });
  it('Should match snapshot', () => {
    const renderedComponent = renderer.create(
      <CollapseHeader>Test</CollapseHeader>,
    );
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
  it('Should render children correctly', () => {
    const renderedComponent = renderer.create(
      <CollapseHeader>
        <Text>test</Text>
      </CollapseHeader>,
    );
    expect(renderedComponent.root.props.children.type).toBe(Text);
    expect(renderedComponent.root.props.children.props.children).toBe('test');
  });
  it('Should have same type when wrapped inside a view', () => {
    const renderedComponent = renderer.create(
      <View>
        <CollapseHeader>Test</CollapseHeader>
      </View>,
    );
    expect(renderedComponent.root.props.children.type).toBe(CollapseHeader);
  });
});
