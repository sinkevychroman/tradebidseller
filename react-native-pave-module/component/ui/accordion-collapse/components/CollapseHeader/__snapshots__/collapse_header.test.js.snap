// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Header Should match snapshot 1`] = `
<View>
  Test
</View>
`;
